### PR TITLE
Stepper: Add unified domain step

### PR DESCRIPTION
**Pull request by alshakero**
Context: pdDR7T-1Ed-p2#comment-2179

## Proposed Changes

This aims to create the thinnest wrapper possible to wrap the Domains step in /start and have it in Stepper as well, to achieve parity without repeating any work. 

## Why are these changes being made?
To be able to delete `client/landing/stepper/declarative-flow/internals/steps-repository/domains` and `client/landing/stepper/declarative-flow/internals/steps-repository/choose-a-domain` and have a single Domains step across all frameworks and places. 

It's still having an issue of handling the picked domain. Needs more work. 

## Testing Instructions

1. Go to /setup/onboarding.
2. Play with the domain step.
3. It should be identical to the one at /start/domains.
4. It should allow selecting multiple or one domain domains.
5. The flow shouldn't take you to checkout if you pick a free domain.
6. The flow SHOULD take you to checkout if you pick one or more paid domains.
7. It should look identical on mobile too.

### Back button
1. If you go to the /setup/onboarding flow without owning any sites, there shouldn't be back button in the domains step.
2. If you go to  /setup/onboarding while owning one site, the back button should say "Back to My Home".
3. If you go to /setup/onboarding while owning two or more sites, back button should say "Back to sites" and takes you to /sites.

### Regression testing
1. Go to https://wpcalypso.wordpress.com/setup/site-setup/goals?siteSlug=YOUR_SITE.
2. Make sure the back button works great.

### What's not to test

1. The interaction with the plans grid. The plans grid isn't yet compatible with this domain step, so it doesn't know if you picked a paid domain or such. Don't test the plans grid, just the domains step, please.

## Next steps

1. ~~Find a way to store the selected domain in the Onboarding store in away that accessible to the plans grid. Similar to `client/landing/stepper/declarative-flow/internals/steps-repository/domains`~~ Done.
5. ~~Find a typed way to persist the [step state](https://github.com/Automattic/wp-calypso/blob/add/unified-domain-step/client/landing/stepper/declarative-flow/internals/steps-repository/unified-domains/index.tsx#L96).~~ Done. 
9. Test test test.
10. ~~Consider doing the same thing for the plans step.~~ Done.

### CODE REVIEW

File: client/landing/stepper/declarative-flow/internals/steps-repository/create-site/index.tsx [hunk 1]
```diff
@@ -223,6 +240,14 @@ const CreateSite: Step = function CreateSite( { navigation, flow, data } ) {
 			await addProductsToCart( site.siteSlug, flow, productCartItems );
 		}
 
+		if ( domainCartItems?.length && site?.siteSlug ) {
+			await addProductsToCart( site.siteSlug, flow, productCartItems );
+		}
+
+		if ( productCartItems?.length && site?.siteSlug ) {
```
Comment (agrullon95) [inline client/landing/stepper/declarative-flow/internals/steps-repository/create-site/index.tsx]: We can remove this condition. Its a duplicate of lines 239 - 241

**Comment by agrullon95** [comment]
Some testing and questions:

* Can't consistently replicate but when I select a paid domain and plan, I get navigated to `/home` instead of checkout.
* Spacing issues on mobile (search bar, filter button, `Show more results` button, content below `Show more results` button).
* Difference in highlight colors on search and filter buttons. When pressed on mobile, the filter button has a pink highlight (pink for my selected theme).
* Are the domains selected in the domains step and the cart linked? Should removing it from the cart affect if it is selected on the domains step? If I select a domain and select a free plan, should the domain be added to the cart?

File: packages/onboarding/src/step-container/index.tsx [hunk 2]
```diff
@@ -174,6 +180,21 @@ const StepContainer: React.FC< Props > = ( {
 		'is-extra-wide-layout': isExtraWideLayout,
 	} );
 
+	let headers = null;
+	if ( formattedHeader ) {
+		headers = formattedHeader;
+	} else if ( headerText ) {
+		headers = (
+			<FormattedHeader
+				id="domains-header"
```
Comment (heavyweight) [inline packages/onboarding/src/step-container/index.tsx]: Maybe its better to pass formattedHeader from the domains step

Comment (alshakero) [reply packages/onboarding/src/step-container/index.tsx]: That would solve 500 problems and make this PR 10 files smaller. But I didn't want to change the original domain step at all, to ship with zero worries.

Comment (alshakero) [reply packages/onboarding/src/step-container/index.tsx]: Decided to go for it.

**Comment by renancarvalho** [comment]
Missing borders on mobile

image

**Comment by escapemanuele** [comment]
As @renancarvalho, I see some problems on mobile for domains

image


Also, `/start/domains` has `< Back to sites`, while it is missing in this PR.

image

**Comment by escapemanuele** [comment]
I see `Use a domain I own` redirecting to `/start/domains/use-your-domain`, this has to be ported into Stepper, right?

**Comment by escapemanuele** [comment]
In Stepper

image

In Start

image

**Comment by escapemanuele** [comment]
`Show more results` presents this border in Stepper

image

**Comment by escapemanuele** [comment]
Thanks @alshakero, I see many things have been solved! I still see these:

- `Use a domain I own` brings me here

image


============

In Stepper

image

In Start
 
image

============

Still missing the `US` in prices

**Comment by renancarvalho** [comment]
Almost there. 

I noted these issues.

Functionality
 

1. When clicking on “Use a domain I own” It is not redirecting to the proper page/ rendering the proper component
2. Message about already registered domain not present.

Production
image

Localhost
image





Styling.

1. I see div right below `id="wpcom"` the  `.step-route onboarding domains` has padding like 60px 0 0 0 and in production we have 48px 0 0 0.

4. Div with class `domains__step-content domains__step-content-domain-step` needs to remove the padding

5. The subtitles are not using the same font-family.
6. Mobile back button is present and in prod it is not.

**Pull request merged**